src/codegen/expression.rs

```diff
@@ -3224,7 +3253,58 @@ pub(crate) fn log_runtime_error(
 ) {
     if report_error {
         let error_with_loc = error_msg_with_loc(ns, reason, Some(reason_loc));
-        let expr = string_to_expr(error_with_loc);
+        let expr = string_to_expr(error_with_loc + ",\n");
         cfg.add(vartab, Instr::Print { expr });
     }
 }
+
+fn add_prefix_and_delimiter_to_print(mut expr: Expression) -> Expression {
+    let prefix = "print: ".as_bytes();
+    let delimiter = ",\n".as_bytes();
+
+    if let Expression::FormatString(loc, args) = &mut expr {
+        let mut new_vec = Vec::new();
+        new_vec.push((
+            FormatArg::StringLiteral,
+            Expression::BytesLiteral(
+                Loc::Codegen,
+                Type::Bytes(prefix.len() as u8),
+                prefix.to_vec(),
+            ),
+        ));
+        new_vec.append(args);
+        new_vec.push((
+            FormatArg::StringLiteral,
+            Expression::BytesLiteral(
+                Loc::Codegen,
+                Type::Bytes(delimiter.len() as u8),
+                delimiter.to_vec(),
+            ),
+        ));
+
+        Expression::FormatString(*loc, new_vec)
+    } else {
+        Expression::FormatString(
+            Loc::Codegen,
+            vec![
+                (
+                    FormatArg::StringLiteral,
+                    Expression::BytesLiteral(
+                        Loc::Codegen,
+                        Type::Bytes(prefix.len() as u8),
+                        prefix.to_vec(),
+                    ),
+                ),
+                (FormatArg::Default, expr),
+                (
+                    FormatArg::StringLiteral,
+                    Expression::BytesLiteral(
+                        Loc::Codegen,
+                        Type::Bytes(delimiter.len() as u8),
+                        delimiter.to_vec(),
```

Comment:
Here you should create a new string which is `prefix` and `delimiter` concatenated, and then create a single `FormatArg::StringLiteral` rather than two.

Reply:
Please ignore, sorry